### DISCUSSION TITLE
Moose environment update

### DISCUSSION THREAD

**Discussion by joe61vette** [idaholab/moose#22158] [Q&A Getting Started] [answered]
Hello:

I have tried both "conda update --all" and "mamba update --all" but don't get the latest version.  When I use "mamba list", it shows:

```
moose-libmesh             2022.06.06              build_0    https://conda.software.inl.gov/public
moose-libmesh-vtk         9.1.0            py39hc1bf809_7    https://conda.software.inl.gov/public
moose-mpich               4.0.1                   build_2    https://conda.software.inl.gov/public
moose-petsc               3.16.5                  build_3    https://conda.software.inl.gov/public
moose-tools               2022.07.18       py39h5a31bc0_0    https://conda.software.inl.gov/public
```
Interestingly, when I issue the update command, a question mark appears in the folder icon at the top of the terminal window.  I have the following at the bottom of my .zshrc file:
```
# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/Users/joe/mambaforge3/bin/conda' 'shell.zsh' 'hook' 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/Users/joe/mambaforge3/etc/profile.d/conda.sh" ]; then
        . "/Users/joe/mambaforge3/etc/profile.d/conda.sh"
    else
        export PATH="/Users/joe/mambaforge3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda initialize <<<

mamba activate moose

export HOST=$(hostname)
```
Before issuing the update command, the folder icon path shows as: "JMKs MBP/Macintosh HD/Users/joe".  But after issuing either the conda or mamba update command, it changes to: "x86_64-apple-darwin13.4.0//Users/joe".

Thanks for the help,
  Joe Kelly

**Comment by GiudGiud** [comment, marked as answer]
Hi Joe

Never seen the issue with the folder icon.

When conda/mamba fails to update (mamba is a little better at the dependency resolution), it's worth deleting the environment and re-cloning it. If you still get the same packages that are lagging behind, the next culprit is usually the python version in the environment

Please let us know how it goes,
Guillaume

**Reply by joe61vette** [reply]
Thanks Guillaume.  Deleting the environment and reloading it did the trick.  I now have:

moose-libmesh             2022.09.09              build_0    https://conda.software.inl.gov/public
moose-libmesh-vtk         9.1.0               he71d177_12    https://conda.software.inl.gov/public
moose-mpich               4.0.2                   build_2    https://conda.software.inl.gov/public
moose-petsc               3.16.6                  build_1    https://conda.software.inl.gov/public
moose-tools               2022.07.18      py310hcffbc79_0    https://conda.software.inl.gov/public

That I believe are the most recent.

Thanks again,
  Joe

**Reply by GiudGiud** [reply]
same as me
```
moose-libmesh             2022.09.09              build_0    https://conda.software.inl.gov/public
moose-libmesh-vtk         9.1.0               h3f5ae15_12    https://conda.software.inl.gov/public
moose-mpich               4.0.2                   build_2    https://conda.software.inl.gov/public
moose-petsc               3.16.6                  build_1    https://conda.software.inl.gov/public
moose-tools               2022.07.18      py310ha2ead4d_0    https://conda.software.inl.gov/public
```
